compiler/rustc_interface/src/queries.rs

```diff
@@ -107,3 +107,25 @@ impl Linker {
         codegen_backend.link(sess, codegen_results, self.metadata, &self.output_filenames)
     }
 }
+
+/// Checks if the given defid refers to an item that is unnamable, such as one defined in a const block.
```

Comment:
What do you think about:

`````suggestion
/// Checks if the given `DefId` refers to an unnamable item like:
///
/// ```
/// pub fn f() {
///     // In here, `X` is not reachable outside of `f`, making it unnamable.
///     pub struct X;
/// }
/// ```
`````

Reply:
I'd advise to put the doc comment in both places then. :)

Also, I don't have an IDE providing doc comments for an item (which sounds really nice) but I don't think it's an argument in favor of having shorter doc comments. More like a bug to report on the IDE.

Reply:
I'm not advising for shorter docs, I'm just trying to follow [the official style guide for docs](https://doc.rust-lang.org/nightly/rustdoc/how-to-write-documentation.html#documenting-components).

The point is the first sentence needs to make sense on its own.

Reply:
> I'm not advising for shorter docs, I'm just trying to follow [the official style guide for docs](https://doc.rust-lang.org/nightly/rustdoc/how-to-write-documentation.html#documenting-components).

Which I co-wrote. I'm ashamed. You're absolutely right.

What about:

```suggestion
/// Checks if the given `DefId` refers to an unnamable item, such as one defined in a function.
///
/// Example:
///
/// ```
/// pub fn f() {
///     // In here, `X` is not reachable outside of `f`, making it unnamable.
///     pub struct X;
/// }
/// ```
```

Reply:
That's incredibly similar to what I just pushed, curious if you think that's good enough.